### ISSUE TITLE
Update cache when content changes

### ISSUE THREAD

**Issue by Mecanik** [stephan13360/cloudflare-worker#2]
Amazing script! I was working for a while on this and you saved me a lot of time (and reminded me about what I forgot regarding cookies etc...).

However, the biggest challenge I see is how to update the content when it changes.

Any ideas?

Thanks!

**Comment by stephan13360** [comment]
Hi Mecanik,

the cache is refreshed in the background on every request. Only the first request after content changes will show the old, cached version and the second and all later request will then show the updated content.

**Comment by Mecanik** [comment]
Thanks, and sorry for I cannot see where you check and update the cache. I believe there should be a case of: "if content != cachedContent = update cache and serve new cache".

**Comment by stephan13360** [comment]
Have you read the explanation of the worker? This should explain it.
https://www.sherbers.de/cache-html-with-cloudflare-workers/

TLDR: The getOrigin() function is responsible for contacting the origin on every request and saving the response to the Cloudflare cache. I don't check if the content changed, I just send it to the cache regardless, overwriting it every time: `event.waitUntil(cache.put(cacheRequest, originResponse.clone()));`

**Comment by Mecanik** [comment]
Ah, I see now... thanks. Well that kind of defeats the purpose when you have a slow responding origin. I think I will improve your script a bit to what I mentioned, perhaps by checking the content-length header or something similarly fast.

Thanks!

**Comment by stephan13360** [comment]
A slow responding origin is not a problem, the worker does not care if the origin even answers at all. The worker will serve the content from cache immediately, way before the origin answered. The worker will then keep running after having served the cache to the client and update the cache when the origin sends its response. I can recommend reading the blog post if you want to know what I intended this worker to do.

**Comment by Mecanik** [comment]
Thanks, I will do just free up some time. Also, what you mentioned there: "The worker will then keep running after having served the cache to the client and update the cache when the origin sends its response." is bad from my opinion because of two reasons:

1. A "hang" running worker after the page was served?
2. You already served the "old" cached response and the new content will become visible to the user only he refreshed the page.

**Comment by Mecanik** [comment]
Indeed you are correct, I just saw https://developers.cloudflare.com/workers/examples/cache-api/. I shall test your script asap :)

Thanks!

**Comment by Mecanik** [comment]
Seems to work nicely so far! Many thanks.

**Comment by stephan13360** [comment]
No problem. Have fun with it.
